python/mxnet/ndarray/numpy/linalg.py

```diff
@@ -18,5 +18,185 @@
+    array(7.)
+    >>> np.linalg.norm(a, -1)
+    array(0.)
+    >>> np.linalg.norm(b, -1, axis=(0, 1))
+    array(6.)
+    >>> np.linalg.norm(a, 2)
+    array(7.745967)
+
+    >>> np.linalg.norm(a, -2)
+    array(0.)
+    >>> np.linalg.norm(a, 3)
+    array(5.8480353)
+    >>> np.linalg.norm(a, -3)
+    array(0.)
+
+    Using the `axis` argument to compute vector norms:
+
+    >>> c = np.array([[ 1, 2, 3],
+    ...               [-1, 1, 4]])
+    >>> np.linalg.norm(c, axis=0)
+    array([1.4142135, 2.236068 , 5.       ])
+    >>> np.linalg.norm(c, axis=1)
+    array([3.7416573, 4.2426405])
+    >>> np.linalg.norm(c, ord=1, axis=1)
+    array([6., 6.])
+
+    Using the `axis` argument to compute matrix norms:
+
+    >>> m = np.arange(8).reshape(2,2,2)
+    >>> np.linalg.norm(m, axis=(1,2))
+    array([ 3.7416573, 11.224973 ])
+    >>> np.linalg.norm(m[0, :, :]), np.linalg.norm(m[1, :, :])
+    (array(3.7416573), array(11.224973))
+
+    """
+    if ord == 'nuc':
+        raise NotImplementedError("Norm does not support nuclear norm.")
+    if axis is None or isinstance(axis, (int, tuple)):  # pylint: disable=too-many-nested-blocks
+        if axis is not None:
+            if isinstance(axis, int):
+                axis = (axis, )
+            if len(axis) == 2:
+                if ord in ['inf', '-inf']:
+                    row_axis, col_axis = axis
+                    if not keepdims:
+                        if row_axis > col_axis:
+                            row_axis -= 1
+                    if ord == 'inf':
+                        return _mx_nd_np.sum(_mx_nd_np.abs(x), axis=col_axis, keepdims=keepdims).max(axis=row_axis, keepdims=keepdims)  # pylint: disable=line-too-long
+                    else:
+                        return _mx_nd_np.sum(_mx_nd_np.abs(x), axis=col_axis, keepdims=keepdims).min(axis=row_axis, keepdims=keepdims)  # pylint: disable=line-too-long
+                if ord in [1, -1]:
+                    row_axis, col_axis = axis
+                    if not keepdims:
+                        if row_axis < col_axis:
+                            col_axis -= 1
+                    if ord == 1:
+                        return _mx_nd_np.sum(_mx_nd_np.abs(x), axis=row_axis, keepdims=keepdims).max(axis=col_axis, keepdims=keepdims)  # pylint: disable=line-too-long
+                    elif ord == -1:
+                        return _mx_nd_np.sum(_mx_nd_np.abs(x), axis=row_axis, keepdims=keepdims).min(axis=col_axis, keepdims=keepdims)  # pylint: disable=line-too-long
+        if ord == 'inf':
+            return _npi.norm(x, ord=2, axis=axis, keepdims=keepdims, flag=3)
+        elif ord == '-inf':
+            return _npi.norm(x, ord=2, axis=axis, keepdims=keepdims, flag=4)
+        elif ord is None:
+            return _npi.norm(x, ord=2, axis=axis, keepdims=keepdims, flag=0)
+        elif ord == 2:
+            return _npi.norm(x, ord=2, axis=axis, keepdims=keepdims, flag=-1)
+        elif ord == 'nuc':
```

Comment:
Line 161 already eliminates the case for `nuclear norm`, is this line still necessary?

Reply:
yes, it's necessary,  bacaue if i want to add nuclear norm, i will use it,  you can think it is  repeating。